### DISCUSSION TITLE
Unable to run segmenter code in wasm

### DISCUSSION THREAD

**Discussion by riajain0412** [unicode-org/icu4x#2852] [Q&A]
Hi,

I was trying to run segmenter code on wasm. Below is the code in my rust file:

```
use wasm_bindgen::prelude::*;
use icu_segmenter::LineSegmenter

#[wasm_bindgen]
extern {
    pub fn alert(s: &str);
}

#[wasm_bindgen]
pub fn greet() {
    let segmenter =
    LineSegmenter::try_new_unstable(&icu_testdata::unstable())
        .expect("Data exists");

let breakpoints: Vec<usize> =
    segmenter.segment_str("Hello World").collect();
assert_eq!(&breakpoints, &[6, 11]);
}
```

Code in my Cargo.toml file:

``` 
[package]
name = "hello-wasm"
version = "0.1.0"
authors = ["Ria Jain <riajain20869@gmail.com>"]
description = "A sample project with wasm-pack"
license = "MIT/Apache-2.0"
repository = "https://github.com/yourgithubusername/hello-wasm"
edition = "2018"

[lib]
crate-type = ["cdylib"]

[dependencies]
icu = "1.0.0"
wasm-bindgen = "0.2"
icu_segmenter = "0.7.0"
icu_testdata = "1.0.0"
```

First I run icu4x-datagen command to build data blob and then I run cargo run, it says no `LineSegmenter` in the root

**Comment by sffc** [comment]
Please enable the `icu_segmenter` feature on `icu_testdata`.

**Reply by riajain0412** [reply]
Hi @sffc , How to do this?

**Reply by sffc** [reply]
In your Cargo.toml,

`icu_testdata = { version = "1.0.0", features = ["icu_segmenter"] }`

**Reply by riajain0412** [reply]
still getting the same error

**Reply by sffc** [reply]
Your code looks fine, so long as you make sure it compiles (no syntax errors). `icu_segmenter::LineSegmenter` is an unconditional export.

**Reply by riajain0412** [reply]
This error is coming in wasm-pack command. I'm unable to get that what am I doing wrong.
And also, do I have to push my code in git?

**Reply by sffc** [reply]
It sounds like your issue is likely not ICU4X-specific.

I always encourage building a Rust-only package first, before adding in complexity like wasm-pack.

**Reply by riajain0412** [reply]
I was trying to implement segmenter in Mac also but was facing this issue there as well.
